bin/desi_proc

```diff
@@ -376,10 +385,27 @@ if rank == 0:
                 raise IOError("no {}".format(psfnightfile))
             input_psf[camera] = psfnightfile
         else :
-            # look for a psfnight psf
+            # look for a psfnight psf 
             psfnightfile = findfile('psfnight', args.night, args.expid, camera)
             if os.path.isfile(psfnightfile) :
                 input_psf[camera] = psfnightfile
+            elif args.most_recent_calib:
+                # Set the time as Noon on the day in question
+                today = time.strptime('{} 12'.format(args.night),'%Y%m%d %H')
+                one_day = 60*60*24 # seconds
+                test_night_struct = today
+                successful = False
+                # Search a month in the past
+                for daysback in range(30):
+                    test_night_struct = time.strptime(time.ctime(time.mktime(test_night_struct)-one_day))
+                    test_night_str = time.strftime('%Y%m%d', test_night_struct)
+                    psfnightfile = findfile('psfnight', test_night_str, args.expid, camera)
+                    if os.path.isfile(psfnightfile) :
+                        input_psf[camera] = psfnightfile
+                        successful = True
+                        break
+                if not successful:
+                    input_psf[camera] = findcalibfile([hdr, camhdr[camera]], 'PSF')
```

Comment:
This logic looks good but we'll need similar logic starting around line 744 for finding the most recent fiberflatnight too.  Let's pull this lookback code into a function so that it can look back in time for either a psfnight or a fiberflatnight.